instrumentation/aws_lambda/lib/opentelemetry/instrumentation/aws_lambda/handler.rb

```diff
@@ -0,0 +1,172 @@
+# frozen_string_literal: true
+
+# Copyright The OpenTelemetry Authors
+#
+# SPDX-License-Identifier: Apache-2.0
+
+module OpenTelemetry
+  module Instrumentation
+    module AwsLambda
+      # Handler class that creates a span around the _HANDLER
+      class Handler
+        attr_reader :handler_method, :handler_class
+
+        # anytime when update the code in a Lambda function or change the functional configuration,
+        # the next invocation results in a cold start; therefore these instance variable will be up-to-date
+        def initialize
+          @flush_timeout    = ENV.fetch('OTEL_INSTRUMENTATION_AWS_LAMBDA_FLUSH_TIMEOUT', '30000').to_i
+          @original_handler = ENV['ORIG_HANDLER'] || ENV['_HANDLER'] || ''
+          @handler_class    = nil
+          @handler_method   = nil
+          @handler_file     = nil
+
+          resolve_original_handler
+        end
+
+        # We want to capture the error if user's handler is causing issue
+        # but our wrapper and handler shouldn't cause any issue
+        def call_wrapped(event:, context:)
+          parent_context   = extract_parent_context(event)
+          span_attributes  = otel_attributes(event, context)
+
+          response = nil
+          original_handler_error = nil
+          begin
+            response = call_original_handler(event: event, context: context)
```

Comment:
@xuan-cao-swi question: it looks like we're calling the wrapped (actual) function handler here, but it's only afterwards in line 45 that we're starting a span for the function? If that's true it is counter to what I expect an instrumenter to do--it should capture the timing of the actual traced operation, the function handler execution in this case.

Reply:
Yes, after double check with python instrumentation, the span time should include the execution of wrapped function. Updated.